src/common/matching/tagMatcher.ts

```diff
@@ -104,6 +132,27 @@ export class TagMatcher {
     }
   }
 
+  /**
+   * Most of the time the relevant tag is the innermost tag, but when Visual mode is active,
+   * the rules are different.
+   * When the cursorStart is standing on the < character of the inner tag, with "at" we must
+   * jump to the outer tag.
+   */
+  determineRelevantTag(
+    tagsSurrounding: MatchedTag[],
+    adjustedStartPosOffset: number,
+    selectionActive: boolean
```

Comment:
Is there a reason for using this `selectionActive` flag rather than checking the mode directly?

Reply:
Hm..you mean just checking whether `vimState.currentMode === Mode.Visual`? If so, the reason is that in Vim if you enter visual mode but don't select anything (i.e. `startCursorPosition === endCursorPosition`), the `at` motion will return the inner tag. If we were to compare modes here instead of cursor positions, `vat` would select the outer tag because of the mode. I apologize, I should have added a comment to specify that edge case.